src/header.wppl

```diff
@@ -469,11 +469,48 @@ var OptimizeThenSample = function(wpplFn, options) {
   return SampleGuide(wpplFn, opts);
 };
 
-var Infer = function(options, maybeFn) {
-  if (!util.isObject(options)) {
-    util.fatal('Infer: expected first argument to be an options object.');
+/*
+ * DefaultInfer() called when no options are specified
+ * var enumTol: upper bound for enumeration tree size, enumeration ends when above threshold.
+ * var minSampleRate: lower bound for #sample/sec, rejection ends when below threshold.
+ * var sampleNum: number of samples/particles used for default rejection/SMC/MCMC.
+ * the "probe" field in the option is a signal to enumerate/rejection/SMC
+ * in probe mode these methods don't throw error, but return useful information for deciding
+ * next step in the heuristics flowchart
+ */
+var DefaultInfer = function(wpplFn, options) {
+  var enumTol = 200000;
+  var minSampleRate = 250; //from 1 to sampleNum
+  var sampleNum = 1000;
+  var enumResult = Enumerate(wpplFn, {probe: enumTol, strategy: 'depthFirst'});
+  if (_.isObjectLike(enumResult)) {
```

Comment:
Could we use `dists.isDist()` to check the return value here? (It might make this a little easier to understand.)